src/models/subject.ts

```diff
@@ -0,0 +1,157 @@
+import { Classification } from "./decisions/classification";
+import { IDecision, VerificationDecision } from "./decisions/decision";
+import { Verification } from "./decisions/verification";
+
+export type SubjectData = Record<string, unknown>;
+
+export interface ISubjectWrapper {
+    subject: Subject;
+    identifier: symbol;
+    clientCached: boolean;
+    serverCached: boolean;
+
+    url: string;
+    decisions: Map<symbol, IDecision[]>;
+    tag: string;
+}
+
+
+export interface Tag {
+    id?: number;
+    text: string;
+}
+
+export class Subject {
+    public constructor(data: SubjectData) {
+        this.data = data;
+        this.identifier = Symbol("verification-subject");
+    }
+
+    public identifier: symbol;
+    public data: SubjectData;
+}
+
+// the Verification model is emitted by the oe-verification-grid as a CustomEvent
+export class SubjectWrapper implements ISubjectWrapper {
+    public constructor(data: ISubjectWrapper) {
+        this.subject = data.subject;
+        this.decisions = data.decisions ?? new Map();
+        this.url = data.url;
+        this.tag = data.tag;
+
+        this.identifier = Symbol("decision");
+    }
+
+    // aka: context
+    // this is the native data model used by the host application
+    // or this could be the csv row
+    public identifier: symbol;
```

Comment:
hangon, didn't subject have an identifier?

why does the wrapper need an identifier? what purpose does it serve?

Reply:
`SubjectWrapper` used to be `DecisionWrapper` a few commits back and was used to add/remove decisions

The identifier is no longer needed